test-data/unit/check-classes.test

```diff
@@ -281,10 +281,9 @@ class A:
     def g(self) -> 'A': pass
 class B(A):
     def f(self) -> A: pass  # Fail
-    def g(self) -> None: pass  # Fail
+    def g(self) -> None: pass
 [out]
 main:6: error: Return type of "f" incompatible with supertype "A"
-main:7: error: Return type of "g" incompatible with supertype "A"
```

Comment:
Well, only without `--strict-optional`, or if the base class has `-> Optional[A]`.